build fail. configparser contain c99 style.

### environment
CentOS 7.9
NSD 4.6.0

### command
git checkout NSD_4_6_0_REL
autoreconf
./configure --with-libevent --with-user=nsd --with-pidfile=/var/run/nsd/nsd.pid
make

### make error
```
./configparser.y: In function ‘c_parse’:
./configparser.y:1062:7: error: ‘for’ loop initial declarations are only allowed in C99 mode
       for(struct component *i = $2; i; i = i->next) {
       ^
./configparser.y:1062:7: note: use option -std=c99 or -std=gnu99 to compile your code
./configparser.y:1069:33: error: redefinition of ‘i’
       for(struct component *j, *i = $2; i; i = j) {
                                 ^
./configparser.y:1062:29: note: previous definition of ‘i’ was here
       for(struct component *i = $2; i; i = i->next) {
                             ^
./configparser.y:1069:7: error: ‘for’ loop initial declarations are only allowed in C99 mode
       for(struct component *j, *i = $2; i; i = j) {
       ^
make: *** [configparser.o] Error 1
```

I look #71 remove c99 style code.
or c99 enable autoreconf.

--- a/configure.ac
+++ b/configure.ac
@@ -174,7 +174,7 @@ AC_ARG_WITH([user],
 AC_SUBST(user)
 AC_DEFINE_UNQUOTED(USER, ["$user"], [the user name to drop privileges to])
 
-AC_PROG_CC
+m4_version_prereq([2.70], [AC_PROG_CC], [AC_PROG_CC_STDC])
 AC_PROG_SED
 AC_PROG_AWK
 AC_PROG_GREP

--- a/doc/RELNOTES
+++ b/doc/RELNOTES
@@ -7,6 +7,7 @@ BUG FIXES:
 	- Fix static analyzer reports, fix wrong log print when skipping xfr,
 	  fix to print error on pipe read fail, and assert an xfr is in
 	  progress during packet checks.
+	- Use AC_PROG_CC_STDC with autoconf versions prior to 2.70.
 
 4.6.0
 ================

